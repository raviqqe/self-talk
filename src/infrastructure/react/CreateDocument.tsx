import { useState } from "react";
import { MdAdd } from "react-icons/md";
import styled from "styled-components";
import { CircleButton } from "./CircleButton";
import { MarkdownTextArea } from "./MarkdownTextArea";
import { InsertFilesFunction } from "./utilities";

const Container = styled.div`
  display: flex;
  align-items: center;
  padding: 1em;
`;

const StyledMarkdownTextArea = styled(MarkdownTextArea)`
  margin-right: -1em;
  max-height: 80vh;
`;

interface IProps {
  createDocument: (text: string) => Promise<void>;
  insertFiles: InsertFilesFunction;
}

export const CreateDocument = ({
  createDocument,
  insertFiles,
}: IProps): JSX.Element => {
  const [text, setText] = useState("");
  const onSubmit = async (): Promise<void> => {
    setText("");
    await createDocument(text);
  };

  return (
    <Container>
      <StyledMarkdownTextArea
        insertFiles={insertFiles}
        onSubmit={onSubmit}
        setText={setText}
        text={text}
      />
      <CircleButton aria-label="Create" onClick={onSubmit}>
        <MdAdd />
      </CircleButton>
    </Container>
  );
};
