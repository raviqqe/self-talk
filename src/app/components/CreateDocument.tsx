import { styled } from "@linaria/react";
import { type JSX, useState } from "react";
import { MdAdd } from "react-icons/md";
import { documentCreator } from "../../main/document-creator.js";
import { CircleButton } from "./CircleButton.js";
import { MarkdownTextArea } from "./MarkdownTextArea.js";

const Container = styled.div`
  display: flex;
  align-items: center;
  padding: 1em;
`;

const StyledMarkdownTextArea = styled(MarkdownTextArea)`
  max-height: 80vh;
`;

const StyledCircleButton = styled(CircleButton)`
  margin-left: -0.5rem;
`;

interface Props {
  className?: string;
}

export const CreateDocument = ({ className }: Props): JSX.Element => {
  const [text, setText] = useState("");
  const onSubmit = async (): Promise<void> => {
    setText("");
    await documentCreator.create(text);
  };

  return (
    <Container className={className}>
      <StyledMarkdownTextArea
        onChange={setText}
        onSubmit={onSubmit}
        text={text}
      />
      <StyledCircleButton aria-label="Create" onClick={onSubmit}>
        <MdAdd />
      </StyledCircleButton>
    </Container>
  );
};
