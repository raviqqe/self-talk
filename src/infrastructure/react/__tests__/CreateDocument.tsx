import { create } from "react-test-renderer";
import { CreateDocument } from "../CreateDocument";

it("renders", () => {
  expect(
    create(
      <CreateDocument
        createDocument={async () => {}}
        insertFiles={async () => "url"}
      />
    ).toJSON()
  ).toMatchSnapshot();
});
