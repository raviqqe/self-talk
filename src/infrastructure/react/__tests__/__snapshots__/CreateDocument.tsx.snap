// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-AxhUy jkncAu"
>
  <textarea
    async={true}
    className="sc-AxiKw emBwCb sc-AxgMl cNIsWQ"
    onChange={[Function]}
    onDrop={[Function]}
    onKeyDown={[Function]}
    onPaste={[Function]}
    placeholder="Write in Markdown ..."
    rows={1}
    value=""
  />
  <button
    aria-label="Create"
    className="sc-AxjAm sc-AxirZ ekcjUO"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </button>
</div>
`;
